### ISSUE TITLE
Add config validation

### ISSUE THREAD

**Issue by jacobtomlinson** [opsdroid/opsdroid#770]
Currently we are not doing a very good job of validating config before loading opsdroid.

An example of this is that we have [very old and now invalid configs](https://github.com/opsdroid/opsdroid/blob/bc95a2bd4d7cf921974f4ee01ec1b51003249bc3/tests/configs/minimal.yaml) in our tests and they are still passing. In that config `skills` is a dictionary, which was changed to a list a long time ago.

We should check our configs are valid before starting to load.

**Comment by bicep** [comment]
Hello, I'd love to help out on this project. Do you think this is a good place to start and if so can you say more about what need to be done for the config tests?

**Comment by FabioRosado** [comment]
Hello @bicep welcome to the project. Can't say exactly if this is the right issue for you - I guess it depends on how well you know yaml and pyyaml 😄 

Basically we are loading the `config.yaml` but there is little validation done on this file (for examples, adding three spaces instead of two will make the config break).


As for the tests, what needs to change is from a dictionary into a list and update them - also checking if things will break or not. 

I hope this will helps a bit. If you think this is a bit out of scope of what you can do you can check the issues with [good first issue](https://github.com/opsdroid/opsdroid/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) or the [intermediate level](https://github.com/opsdroid/opsdroid/issues?q=is%3Aissue+is%3Aopen+label%3Aintermediate).

Let me know if you need any help with anything 😄 👍

**Comment by paw39** [comment]
Hi! I'd like to try fix this issue. Could you tell me a little bit more about this? When we need to check config.yaml ? Any code snippet or hint?

**Comment by paw39** [comment]
I've found some class named Loader, I think it should be there.

**Comment by Cadair** [comment]
We could take a leaf out of Home Assistants book here and have connectors and skills define a schema for their config. 

See https://developers.home-assistant.io/docs/en/development_validation.html and https://pypi.org/project/voluptuous/ is the lib they use.

**Comment by kollyuchka** [comment]
Hello, I would like to handle this issue. Is it still actual?

**Comment by FabioRosado** [comment]
Heya and welcome!

Yeah this issue is still actual and would be great if you could tackle it. We have just updated pyyaml to the latest version 5.1.1(2?) but home assistant is still using 3-4.

Please let me know if you need any help with this issue.

Thank you 😄👍

**Comment by kollyuchka** [comment]
@FabioRosado 
Here is the PR #1003. 
Can you review ?

A lot of tests that deals with old config style fail the validation, so I skipped them.
Is that a correct solution? 
Should we delete the tests?

**Comment by FabioRosado** [comment]
Ah okay thanks for this 😄👍
I am going to work now but will review the PR once I’m home
